From 86389c62eab518683ce29d10a5a33c0e098f6b7c Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Wed, 20 Feb 2019 16:05:18 +0800
Subject: [PATCH 670/706] felix: export felix_ptp_gettime

Exported felix_ptp_gettime for global usage.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
(cherry picked from commit ca9e06d3ed76459a966c07cff34aa02284297e70)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/felix_ptp.c | 3 ++-
 drivers/net/ethernet/mscc/ocelot.h    | 6 ++++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mscc/felix_ptp.c b/drivers/net/ethernet/mscc/felix_ptp.c
index 9a2b1e8125db..6b7112df3fef 100644
--- a/drivers/net/ethernet/mscc/felix_ptp.c
+++ b/drivers/net/ethernet/mscc/felix_ptp.c
@@ -6,7 +6,7 @@
 
 #include "ocelot.h"
 
-static int felix_ptp_gettime(struct ptp_clock_info *ptp, struct timespec64 *ts)
+int felix_ptp_gettime(struct ptp_clock_info *ptp, struct timespec64 *ts)
 {
 	struct ocelot *ocelot = container_of(ptp, struct ocelot, ptp_caps);
 	u32 val, tod_ns, tod_sec_lsb, tod_sec_msb;
@@ -34,6 +34,7 @@ static int felix_ptp_gettime(struct ptp_clock_info *ptp, struct timespec64 *ts)
 
 	return 0;
 }
+EXPORT_SYMBOL_GPL(felix_ptp_gettime);
 
 static int felix_ptp_settime(struct ptp_clock_info *ptp,
 			     const struct timespec64 *ts)
diff --git a/drivers/net/ethernet/mscc/ocelot.h b/drivers/net/ethernet/mscc/ocelot.h
index dfaa2769d43c..a0f378c0f3d5 100644
--- a/drivers/net/ethernet/mscc/ocelot.h
+++ b/drivers/net/ethernet/mscc/ocelot.h
@@ -634,6 +634,7 @@ int ocelot_probe_port(struct ocelot *ocelot, u8 port,
 #ifdef CONFIG_MSCC_FELIX_SWITCH_PTP_CLOCK
 int felix_ptp_init(struct ocelot *ocelot);
 void felix_ptp_remove(struct ocelot *ocelot);
+int felix_ptp_gettime(struct ptp_clock_info *ptp, struct timespec64 *ts);
 #else
 static inline int felix_ptp_init(struct ocelot *ocelot)
 {
@@ -641,6 +642,11 @@ static inline int felix_ptp_init(struct ocelot *ocelot)
 }
 
 static inline void felix_ptp_remove(struct ocelot *ocelot) { }
+static inline int felix_ptp_gettime(struct ptp_clock_info *ptp,
+				    struct timespec64 *ts)
+{
+	return 0;
+}
 #endif
 
 extern struct notifier_block ocelot_netdevice_nb;
-- 
2.17.1

